fluent-bit: add package

rthakur33 opened the pull request:
* add package fluent-bit v3.1.0

krant commented:
There is no need to specify all CMake options matching the defaults - it makes Makefile larger and harder to maintain.

rthakur33 commented:
> There is no need to specify all CMake options matching the defaults - it makes Makefile larger and harder to maintain.

Thanks for your comment, it is addressed now.

BKPepe commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,47 @@
+PKG_VERSION:=3.1.0
+PKG_RELEASE:=1
+
+PKG_SOURCE=v$(PKG_VERSION).tar.gz
```
This is not correct. We can not download files "vVERSION", because in the dl folder it is going to be mess and we would not be sure to which package does it belong, yeah, you can verify it manually, but you probably won't verify it, if it was not overwritten by the other package with the same version. 

So, it should be as usual as it is done in other Makefiles - packageName-Version.

rthakur33 replied:
done

BKPepe commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,47 @@
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/fluent-bit
+	CATEGORY:=Utilities
+	DEPENDS:= +libyaml +libopenssl +libcurl +libatomic +musl-fts +flex +bison
+	TITLE:=Fluent-Bit
+	URL:=https://fluentbit.io/
+endef
```
Two spaces as indentation here.

rthakur33 replied:
done

BKPepe commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,47 @@
+	URL:=https://fluentbit.io/
+endef
+
+define Package/fluent-bit/description
+	Fluent Bit is a super fast, lightweight, and highly scalable logging and metrics processor and forwarder.
+endef
```
Two spaces here as well.

rthakur33 replied:
done

BKPepe commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,47 @@
+	-DFLB_WASM=No \
+	-DFLB_LUAJIT=No
+
+define Package/fluent-bit/install
+	$(INSTALL_DIR) $(1)/usr/sbin
+	$(INSTALL_DIR) $(1)/etc/fluent-bit
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/bin/fluent-bit $(1)/usr/sbin/
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/conf/parsers.conf $(1)/etc/fluent-bit/parsers.conf
+endef
```
We can order it a little bit more.

First folder, then file.

Also, the config file should be INSTALL_CONF and shouldnt it be marked as conffile to prevent to be overwritten by update?

rthakur33 replied:
done

BKPepe reviewed (changes requested):
In the pull request and as well in the commit description, I do miss some description. If the package was compile tested, run tested and also why do we need to add this.

1. This goes into admin folder, but you do have category Utilities in the Makefile.
I suggest to look at it more, where it belongs, because it is also missing Section in the define.

rthakur33 commented:
> In the pull request and as well in the commit description, I do miss some description. If the package was compile tested, run tested and also why do we need to add this.
> 
> 1. This goes into admin folder, but you do have category Utilities in the Makefile.
>    I suggest to look at it more, where it belongs, because it is also missing Section in the define.

Thanks for reviewing this, I have addressed all your comments in the update pull request.

BKPepe commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,55 @@
+PKG_SOURCE_URL:=https://github.com/fluent/fluent-bit.git
+PKG_SOURCE=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_VERSION=v$(PKG_VERSION)
+PKG_MIRROR_HASH:=skip
```
Please don't do this. This is even mentioned in the documentation, how you should not do it. I think this says it all. Skipping something is not wanted not even for checking checksum. Not sure, where did you find it.

rthakur33 replied:
sorry, updated

BKPepe commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,55 @@
+PKG_VERSION:=3.1.0
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/fluent/fluent-bit.git
+PKG_SOURCE=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_VERSION=v$(PKG_VERSION)
```
Here you are mixing at least it looks like two things. Checking git sources and downloading tarball. You should decide which one would you want to use.

rthakur33 replied:
yes, I realise, updated, thanks!

1715173329 commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,54 @@
+define Package/fluent-bit
+  SECTION:=admin
+  CATEGORY:=Administration
+  DEPENDS:= +libyaml +libopenssl +libcurl +libatomic +musl-fts +flex +bison
+  TITLE:=Fluent-Bit
+  URL:=https://fluentbit.io/
```
```suggestion
  TITLE:=Fast and Lightweight Logs and Metrics processor
  URL:=https://fluentbit.io/
  DEPENDS:=+libyaml +libopenssl +libcurl +libatomic +musl-fts +flex +bison
```

rthakur33 replied:
thanks for the review, done

1715173329 commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,54 @@
+endef
+
+define Package/fluent-bit/description
+  Fluent Bit is a super fast, lightweight, and highly scalable logging and metrics processor and forwarder.
```
```suggestion
  Fluent Bit is a super fast, lightweight, and highly scalable logging
  and metrics processor and forwarder.
```

wrap a new line for every ~70 chars

rthakur33 replied:
thanks for reviewing, done.

BKPepe commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,55 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=fluent-bit
+PKG_VERSION:=3.1.0
```
Before merging this, can we update this to 3.1.3, please? :)

BKPepe commented on admin/fluent-bit/Makefile:
```diff
@@ -0,0 +1,55 @@
+/etc/fluent-bit/parsers.conf
+endef
+
+# General options
```
Do we need this comment?
```suggestion
```

BKPepe reviewed (approved):
Otherwise 👍 Thank you!

rthakur33 commented:
> Otherwise 👍 Thank you!

thanks for the review, comments addressed.

BKPepe commented:
Two approvals, two reviewers, so many reviews, no other choice than merging this. :-)

hnyman commented:
Seems to fail in buildbot due to dependencies, as some libraries are found...
(and seems to vary between targets...)

https://downloads.openwrt.org/snapshots/faillogs/aarch64_cortex-a72/packages/fluent-bit/compile.txt

> 
> Package fluent-bit is missing dependencies for the following libraries:
> libz.so.1
> libzstd.so.1

https://downloads.openwrt.org/snapshots/faillogs/mips_24kc/packages/fluent-bit/compile.txt
https://downloads.openwrt.org/snapshots/faillogs/x86_64/packages/fluent-bit/compile.txt
> Package fluent-bit is missing dependencies for the following libraries:
> libsasl2.so.3
> libz.so.1
> libzstd.so.1

krant commented:
Adding this to CMAKE_OPTIONS must fix the issue:
```
	-DWITH_ZLIB=OFF \
	-DWITH_ZSTD=OFF \
	-DWITH_SASL=OFF \
```
